src/compute.py

```diff
@@ -0,0 +1,250 @@
+            the MPCD component.
+
+    """
+
+    def __init__(
+        self,
+        num_bins,
+        lower_bounds,
+        upper_bounds,
+        filter=None,
+        include_mpcd_particles=False,
+    ):
+        super().__init__()
+
+        param_dict = ParameterDict(
+            num_bins=(int, int, int),
+            lower_bounds=(float, float, float),
+            upper_bounds=(float, float, float),
+            filter=OnlyTypes(hoomd.filter.ParticleFilter, allow_none=True),
+            include_mpcd_particles=bool(include_mpcd_particles),
+        )
+        param_dict.update(
+            dict(
+                num_bins=num_bins,
+                lower_bounds=lower_bounds,
+                upper_bounds=upper_bounds,
+                filter=filter,
+            )
+        )
+        self._param_dict.update(param_dict)
+
+    def _attach_hook(self):
+        sim = self._simulation
+
+        cpp_class = getattr(_azplugins, self._make_cpp_class_name())
+
+        num_bins = hoomd._hoomd.uint3()
+        num_bins.x = self.num_bins[0]
+        num_bins.y = self.num_bins[1]
+        num_bins.z = self.num_bins[2]
+
+        lower_bounds = hoomd._hoomd.make_scalar3(
+            self.lower_bounds[0], self.lower_bounds[1], self.lower_bounds[2]
+        )
+        upper_bounds = hoomd._hoomd.make_scalar3(
+            self.upper_bounds[0], self.upper_bounds[1], self.upper_bounds[2]
+        )
+
+        if self.filter is not None:
+            group = sim.state._get_group(self.filter)
+        else:
+            group = None
+
+        self._cpp_obj = cpp_class(
+            sim.state._cpp_sys_def,
+            num_bins,
+            lower_bounds,
+            upper_bounds,
+            group,
+            self.include_mpcd_particles,
+        )
+
+        super()._attach_hook()
+
+    def _make_cpp_class_name(self):
+        cpp_class_name = self.__class__.__name__
+        if isinstance(self._simulation.device, hoomd.device.GPU):
+            cpp_class_name += "GPU"
+        return cpp_class_name
+
+    @property
+    def coordinates(self):
+        """numpy.ndarray: Coordinates of bin centers.
+
+        If binning is performed in more than 1 dimension, a multidimensional
+        array is returned.
+
+        If binning is performed in 1 dimension, a 1 dimensional array is
+        returned.
+
+        """
+        coords = []
+        shape = []
+        for num, lo, hi in zip(self.num_bins, self.lower_bounds, self.upper_bounds):
+            if num > 0:
+                x, dx = numpy.linspace(lo, hi, num, endpoint=False, retstep=True)
+                x += 0.5 * dx
+                coords.append(x)
+                shape.append(num)
+
+        if len(shape) == 0:
+            return None
+
+        if len(shape) > 1:
+            shape.append(len(shape))
+
+        return numpy.reshape(list(itertools.product(*coords)), shape)
+
+    @hoomd.logging.log(category="sequence", requires_run=True)
+    def velocities(self):
```

Comment:
The logging might not work as intended, hoomd gives an error: 

`ValueError: Invalid numpy dimension in gsd log data [log/azplugins/compute/CylindricalVelocityFieldCompute/velocities]`